src/blob/index/core.rs

```diff
@@ -81,14 +84,17 @@ impl<FileIndex: FileIndexTrait> IndexStruct<FileIndex> {
     pub(crate) fn clear(&mut self) {
         self.inner = State::InMemory(BTreeMap::new());
         self.mem = Some(Default::default());
-        self.filter.clear();
+        self.bloom_filter.clear();
+        self.range_filter.clear();
     }
 
-    pub fn check_bloom_key(&self, key: &[u8]) -> Option<bool> {
-        if self.params.filter_is_on {
-            Some(self.filter.contains(key))
+    pub(crate) fn check_filters_key(&self, key: &[u8]) -> bool {
```

Comment:
For readability it is better to introduce `enum` with `NeedAdditionalCheck` and `NotContains` variants. `bool` is not the best type here